spec/facebook_ads/model_spec.rb

```diff
@@ -0,0 +1,220 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe FacebookAds::Model do
+  let(:test_class) do
+    Class.new(described_class) do
+      field :foo
+      field :bar
+    end
+  end
+
+  let(:id) { 42 }
+  let(:test_model) { test_class.new(id) }
+
+  let(:client) { double(FacebookAds::Client) }
+
+  before do
+    allow(FacebookAds).to receive(:client).and_return(client)
+  end
+
+  describe '.field' do
+    it 'creates a round-tripping accessor' do
+      expected = 'foo'
+      test_model.foo = expected
+      expect(test_model.foo).to eql expected
+    end
+  end
+
+  describe '#fetch' do
+    let(:fetch_fields) { ['foo'] }
+    let(:response) { {} }
+    subject { test_model.fetch(fetch_fields) }
+
+    before do
+      allow(client).to receive(:get).and_return(response)
+    end
+
+    it 'queries the data of the correct object' do
+      expect(client).to receive(:get).with(id, anything)
+      subject
+    end
+
+    it 'queries the specified field from the object' do
+      expect(client)
+        .to receive(:get).with(anything, params: { fields: fetch_fields.first })
+      subject
+    end
+
+    describe 'deserialization' do
+      let(:foo_value) { 'A string' }
+      let(:response) { { 'foo' => foo_value } }
+
+      it 'populates the value of foo' do
+        subject
+        expect(test_model.foo).to eql foo_value
+      end
+
+      # Not going to be exhaustive here
+      # but at least check one other type than String
+      context 'value is an integer' do
+        let(:foo_value) { 1337 }
+
+        it 'populates the value of foo' do
+          subject
+          expect(test_model.foo).to eql foo_value
+        end
+      end
+
+      describe 'custom field types' do
+        let(:converted_value) { double('converted value') }
+        let(:converter) { double('Converter', from_json: converted_value) }
+        let(:test_class) do
+          Class.new(described_class) do
+            field :foo, type: :bar
+          end
+        end
+
+        before do
+          allow(FacebookAds::JsonConverter)
+            .to receive(:for).and_return(converter)
+        end
+
+        it 'acquires the correct kind of converter' do
```

Comment:
Nitpicking: The converter being returned is stubbed so it will always return the 'correct' kind of converter. 
Spec could be described as 'queries for correct kind of converter'. Alternatively it is also possible to restrict the stubbing to only return this converter if queried `.with(:bar)`.